lib/query.js

```diff
@@ -4007,12 +4013,83 @@ Query.prototype._findAndModify = function(type, callback) {
  */
 
 function _completeOneLean(doc, res, opts, callback) {
+  if (opts.lean && opts.lean.transform) {
+    for (const key of Object.keys(doc)) {
+      if (Array.isArray(doc[key])) {
+        doc[key].forEach((item) => {
+          for (const k in item) {
+            if (typeof item[k] === 'object' && item[k] != null && Object.keys(item[k]).length) {
+              _completeOneLean(item[k], res, opts);
+            }
+            opts.lean.transform({}, item);
+          }
+        });
+      } else if (typeof doc[key] === 'object' && doc[key] != null && Object.keys(doc[key]).length) {
```

Comment:
This won't work, because we don't want to call `completeOneLean()` on nested paths as opposed to subdocuments.

```javascript
new Schema({
  nestedPath: { prop: String }, // <-- nested path, `nestedPath` isn't a subdocument
  subdocument: new Schema({ prop: String }) // <-- `subdocument` is a subdocument
});
```